pkg/cnb/builder_builder.go

```diff
@@ -83,12 +80,27 @@ func (bb *BuilderBuilder) writeableImage() (v1.Image, error) {
 		layers = append(layers, layer.v1Layer)
 	}
 
-	orderLayer, err := bb.tomlLayer()
+	stackLayer, err := bb.stackLayer()
+	if err != nil {
+		return nil, err
+	}
+
+	orderLayer, err := bb.orderLayer()
 	if err != nil {
 		return nil, err
 	}
 
-	image, err := mutate.AppendLayers(bb.baseImage, append(layers, orderLayer)...)
+	lifecycleLayers, err := bb.lifecycleImage.Layers()
+	if err != nil {
+		return nil, err
+	}
+
+	image, err := mutate.AppendLayers(bb.baseImage, append(lifecycleLayers, stackLayer)...)
+	if err != nil {
+		return nil, err
+	}
+
+	image, err = mutate.AppendLayers(image, append(layers, orderLayer)...)
```

Comment:
Could we combine these, perhaps with a helper?

mutate.AppendLayers(image, append(layers, orderLayer)...)